### DISCUSSION TITLE
Filters for all objects (and not just images)?

### DISCUSSION THREAD

**Discussion by VenkSociety** [fabricjs/fabric.js#6869] [Ideas]
_* Apologies for the poor structure, the only template was for bug reports and seeing as this is not a bug report, it didn't make much sense to follow that._

My suggestion is to allow effects such as sepia, grayscale, tinting, blurring etc. for more than just image elements. Fairly sure it is possible using Canvas' [CanvasRenderingContext2D.filter](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/filter) property. At the moment - to my knowledge - these filters can only be used on image elements _(makes sense considering you have it set up as .image.filters)_ but I think it would be nice and beneficial for a lot of people if it were possible to apply these filters to things such as shapes, drawings, text etc.

**Comment by asturur** [comment]
Ho @derekdinan  the best way to post those ideas is the 'discussions' tab. This is indeed for bug reports.

There is a pr to enable filters for all objects, is unmerged yet. You can have a look in the pr section and try it.
